src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisService.java

```diff
@@ -146,4 +158,62 @@ private Integer getContingenciesCount(List<UUID> ids, UUID networkUuid, String v
+        if (sensitivityAnalysisCsvFileInfos == null ||
+                sensitivityAnalysisCsvFileInfos.getSensitivityFunctionType() == null ||
+                sensitivityAnalysisCsvFileInfos.getResultTab() == null ||
+                CollectionUtils.isEmpty(sensitivityAnalysisCsvFileInfos.getCsvHeaders())) {
+            throw new PowsyblException("Missing information to export sensitivity result as csv : Sensitivity result tab, sensitivity function type and csv file headers must be provided");
+        }
+        ResultsSelector selector = ResultsSelector.builder()
+                .functionType(sensitivityAnalysisCsvFileInfos.getSensitivityFunctionType())
+                .tabSelection(sensitivityAnalysisCsvFileInfos.getResultTab())
+                .build();
+
+        SensitivityRunQueryResult result = getRunResult(resultUuid, selector);
+        if (result == null) {
+            return null;
+        }
+
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            ZipOutputStream zipOutputStream = new ZipOutputStream(outputStream)) {
+            zipOutputStream.putNextEntry(new ZipEntry("sensitivity_result.csv"));
+
+            CsvWriterSettings settings = new CsvWriterSettings();
+            CsvWriter csvWriter = new CsvWriter(zipOutputStream, settings);
+            csvWriter.writeHeaders(sensitivityAnalysisCsvFileInfos.getCsvHeaders());
+            if (selector.getTabSelection() == ResultTab.N) {
+                result.getSensitivities()
+                        .forEach(sensitivity -> csvWriter.writeRow(
+                                sensitivity.getFuncId(),
+                                sensitivity.getVarId(),
+                                nullIfNan(sensitivity.getFunctionReference()),
+                                nullIfNan(sensitivity.getValue())
+                        ));
+            } else if (selector.getTabSelection() == ResultTab.N_K) {
+                result.getSensitivities()
+                        .stream()
+                        .map(SensitivityWithContingency.class::cast)
+                        .forEach(sensitivityWithContingency -> csvWriter.writeRow(
+                                sensitivityWithContingency.getFuncId(),
+                                sensitivityWithContingency.getVarId(),
+                                sensitivityWithContingency.getContingencyId(),
+                                nullIfNan(sensitivityWithContingency.getFunctionReference()),
+                                nullIfNan(sensitivityWithContingency.getValue()),
+                                nullIfNan(sensitivityWithContingency.getFunctionReferenceAfter()),
+                                nullIfNan(sensitivityWithContingency.getValueAfter())
+                        ));
+            }
+
+            csvWriter.close();
+            return outputStream.toByteArray();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
```

Comment:
Idem, we need create a handler RestResponseEntityExceptionHandler and SensibilityAnalysisException as in study-server and security-analysis-server => pass a type FILE_EXPORT_ERROR  and a message. In the handler, return response entity with message